include/eeprom/simulation/eeprom.hpp

```diff
@@ -67,10 +79,154 @@ class EEProm : public I2CDeviceBase,
         write_protected = enabled;
     }
 
+    struct BackingStore {
+        static constexpr std::string_view TEMPFILE_KEY = "<temp file>";
+        static constexpr size_t BACKING_SIZE =
+            static_cast<size_t>(hardware_iface::EEpromMemorySize::ST_16_KBYTE);
+        BackingStore(const BackingStore&) = delete;
+        auto operator=(const BackingStore&) -> BackingStore& = delete;
+        BackingStore(BackingStore&&) = delete;
+        auto operator=(BackingStore&&) -> BackingStore&& = delete;
+        explicit BackingStore(const po::variables_map& variables)
+            : backing(get_prepped_backing_file(variables)) {}
+        ~BackingStore() {
+            if (backing) {
+                if (!std::ferror(backing)) {
+                    std::fflush(backing);
+                }
+                std::fclose(backing);
+            }
+        }
+        static auto get_temp_file() -> std::FILE* {
+            auto tempdir = std::filesystem::temp_directory_path();
+            auto temp_path = tempdir / "eeprom.bin";
+            LOG("Backing up eeprom with tempfile at %s", temp_path.c_str());
+            return std::fopen(temp_path.c_str(), "w+b");
+        }
+        static auto get_backing_file(std::string pathstr) -> std::FILE* {
+            auto path = std::filesystem::path(pathstr);
+            path.make_preferred();
+            LOG("Backing up eeprom with %s", path.c_str());
+            return std::fopen(path.c_str(), "a+b");
+        }
+        static auto get_prepped_backing_file(std::string pathstr)
+            -> std::FILE* {
+            auto file = ((pathstr == TEMPFILE_KEY) ? get_temp_file()
+                                                   : get_backing_file(pathstr));
+            if (!file) {
+                fprintf(stderr, "Could not open backing file %s: %d %s\n",
```

Comment:
This is the only file that writes to `stderr` as far as I can tell, would it make more sense to consolidate all message in the same log output?

Reply:
No, I don't think so - this really doesn't want to be captured because it's not part of the normal logging. This would indicate something going wrong with the simulator itself.